core-kotlin-modules/core-kotlin-collections-6/src/test/kotlin/com/baeldung/parallelOperationsCollections/ParallelOperationCollectionsUnitTest.kt

```diff
@@ -0,0 +1,205 @@
+    @Test
+    fun `using coroutines for parallel operations with Flow`() = runBlocking {
+        logger.info("Using Kotlin Flow")
+        val startTime = Instant.now()
+
+        val filteredPeople = people.asFlow()
+            .flatMapMerge { person ->
+                flow {
+                    emit(
+                        async {
+                            person.setAdult()
+                            person
+                        }.await()
+                    )
+                }
+            }
+            .filter { it.age > 15 }.toList()
+            .sortedBy { it.age }
+
+        startTime.printTotalTime()
+
+        filteredPeople.assertResultsTrue()
+    }
+
+    @Test
+    fun `using RxJava for parallel operations`() { // Observable.class from io.reactivex;
+        logger.info("Using RxJava")
+        val startTime = Instant.now()
+
+        val observable = Observable.fromIterable(people)
+            .flatMap(
+                {
+                    Observable.just(it)
+                        .subscribeOn(Schedulers.computation())
+                        .doOnNext { person -> person.setAdult() }
+                },
+                people.size // Uses maxConcurrency for the number of elements
+            )
+            .filter { it.age > 15 }
+            .toList()
+            .map { it.sortedBy { person -> person.age } }
+            .blockingGet()
+
+        startTime.printTotalTime()
+
+        observable.assertResultsTrue()
+    }
+
+    @Test
+    fun `using RxKotlin for parallel operations`() { // ObservableKt.kt.class from io.reactivex.rxkotlin
+        logger.info("Using RxKotlin")
+        val startTime = Instant.now()
+
+        val observable = people.toObservable()
+            .flatMap(
+                {
+                    Observable.just(it)
+                        .subscribeOn(Schedulers.computation())
+                        .doOnNext { person -> person.setAdult() }
+                },
+                people.size // Uses maxConcurrency for the number of elements
+            ).filter { it.age > 15 }
+            .toList()
+            .map { it.sortedBy { person -> person.age } }
+            .blockingGet()
+
+        startTime.printTotalTime()
+
+        observable.assertResultsTrue()
+    }
+
+    @Test
+    fun `using RxKotlin but still use 1 thread`() { // ObservableKt.kt.class from io.reactivex.rxkotlin
+        logger.info("Using RxKotlin 1 thread")
+        val startTime = Instant.now()
+
+        val observable = people.toObservable()
+            .subscribeOn(Schedulers.io())
+            .flatMap { Observable.just(it) } // Without using the maxConcurrency parameter, so it only uses 1 thread.
+            .doOnNext { person -> person.setAdult() }
+            .filter { it.age > 15 }.toList()
+            .map { it.sortedBy { person -> person.age } }.blockingGet()
+
+        startTime.printTotalTime()
+
+        observable.assertResultsTrue()
+    }
```

Comment:
I'm still not clear about the difference between the `using RxKotlin for parallel operations` and `using RxKotlin but still use 1 thread` examples, there are quite a few different lines in them, not just the maxConcurrency parameter change. Can we either make these the same apart from the maxConcurrency parameter, or explain the difference between the two approaches in the article?

Reply:
CMIIW, So apart from using maxConcurency, we also must to do Schedulers (with subscribeOn()) inside flatMap(). In a flatmap(), we are actually creating observables for a number of elements. Each element we define Schedulers before performing the setAdult() operation.

Meanwhile, if we do subscribeOn(Schedulers.io()) before flatMap(), this will use one thread.

Reply:
What do you think? Is this additional information worth adding to the article?
or just add comments to the code?